### ISSUE TITLE
StrEnum cannot be assigned to Literal

### ISSUE THREAD

**Issue by Bluenix2** [microsoft/pyright#2718]
## Summary

I have a class that inherits from `str` and [`enum.Enum`](https://docs.python.org/3/library/enum.html) but even though it's a clear constant it can't be passed to a literal.

## Reproduceable code

See below:
```python
import enum
from typing import Literal

class StrEnum(str, enum.Enum):
    A = 'a'


def test(a: Literal['a'] = StrEnum.A) -> None:
    ...
```
```
Expression of type "Literal[StrEnum.A]" cannot be assigned to parameter of type "Literal['a']"
  "Literal[StrEnum.A]" cannot be assigned to type "Literal['a']"
```

## Expected behaviour

Since my `StrEnum` class is a subclass of `str` it should have all attributes that `str` does, additionally since it is a constant and will never change it should be able to be applied to the Literal of its value.

**Comment by erictraut** [comment]
From a type checking perspective, I think the current behavior is correct. The type of `StrEnum.A` is `Literal[StrEnum.A]` regardless of the fact that `StrEnum` happens to subclass from `str`. 

As further evidence this is correct, this is the same error mypy provides.

I think you have two options here:

```python
def test(a: Literal["a"] = StrEnum.A.value) -> None:
    ...

def test(a: Literal[StrEnum.A] = StrEnum.A) -> None:
    ...
```

**Comment by Bluenix2** [comment]
> From a type checking perspective, I think the current behavior is correct. The type of `StrEnum.A` is `Literal[StrEnum.A]` regardless of the fact that `StrEnum` happens to subclass from `str`.

That is correct, but what this is about is whether `Literal[StrEnum.A]` is assignable to `Literal[StrEnum.A.value]` (which I think that most people's intuition would say) which I am still not convinced it is not.

> I think you have two options here:
> 
> ```python
> def test(a: Literal["a"] = StrEnum.A.value) -> None:
>     ...
> 
> def test(a: Literal[StrEnum.A] = StrEnum.A) -> None:
>     ...
> ```

Considering how easy the former is this is probably bikeshedding, and I will probably just use that. Although `Union[StrEnum, Literal['a']]` also works.
